File: python_app/de/report/test_report/test_report.py
# Copyright (c) 2013, fefe and contributors
# For license information, please see license.txt

from __future__ import unicode_literals
import frappe
from frappe import _

def execute(filters=None):
	columns, data = [], []

	columns+= [
		{"label":"Name","fieldtype":"Data","fieldname":"name"},
		{"label":_("Test") ,"width":80,"fieldtype": "Data","fieldname":"test"}
	]
	if filters.get("ch"):
		columns+= [{"label":"New","fieldtype":"Date"}]		
	
	doc_data = get_data(filters)


	#row = {"name":"test","test":"sss"}
	#data.append(row)

	for doc in frappe.db.sql("select name , test from tabDashTable"):
		#frappe.msgprint(doc[0])
		row = {"Name":doc[0],"Test":doc[1]} 

		data.append(row)

	return columns, data


def get_data(filters=None):
	#frappe.msgprint(filters.get("test"))
	#myfilters = 'test = %(test)s'
	#myfilters += 'name = %(to_date)s'

	
	data= frappe.db.sql("select name , test from tabDashTable ")

	return data

File: python_app/de/report/test_report/test_report.js
// Copyright (c) 2016, fefe and contributors
// For license information, please see license.txt
/* eslint-disable */

frappe.query_reports["Test REport"] = {
	"filters": [
		{
			"fieldname":"from_date",
			"label": __("From Date"),
			"fieldtype": "Date",
			"default": frappe.datetime.add_months(frappe.datetime.get_today(), -1),
			"reqd": 1
		},
		{
			"fieldname":"to_date",
			"label": __("To Date"),
			"fieldtype": "Date",
			"default": frappe.datetime.get_today(),
			"reqd": 1
		},
		{
			"fieldname":"test",
			"label": __("Test"),
			"fieldtype": "Data"
		},
		{
			"fieldname":"ch",
			"label": __("Active"),
			"fieldtype": "Check",
			on_change: function() {
			//var data = frappe.query_report.get_values().to_date
			var data = frappe.query_report.get_filter_value('to_date');
			frappe.query_report.set_filter_value("test",data);
			frappe.query_report.refersh()
	}
		}
	],
	"onload": function() {
		//frappe.msgprint("Test");
		},
	"formatter": function(value, row, column, data, default_formatter) {
			console.log(data)
			if (data.Test== undefined)
				value="<span style='color:red'>"+value+"</span>"


	return value
				
	}
}
